starlette_zipkin/middleware.py

```diff
@@ -56,6 +67,8 @@ async def dispatch(
 
             finally:
                 reset_root_span(root_span)
+                if not external_tracer:
+                    _tracer_ctx_var.reset(tracer_reset_token)
```

Comment:
I dont understand why when it is usefull. by the way the tracer_reset_token may be unintialized?

Reply:
All this "external" part was to allow for `def tracer(transport)` fixture from conftest.py to take precedence over "internal tracer", otherwise that part wont work as expected.